skbio/stats/distance/_regression.py

```diff
@@ -0,0 +1,170 @@
+from skbio.util._misc import check_random_state
+from skbio.stats.distance._mantel import _order_dms
+
+from skbio.stats.distance import DistanceMatrix
+
+@experimental(as_of="0.4.0")
+def linregress(y, *args, **kwargs):
+    """
+    This module performs a multiple linear regression on distance
+    matrices.  This is done by extracting the upper (or lower)
+    triangular portions of the symmetric distance matrices,
+    and collapsing them to compute a multiple linear regression
+    on the distances.  The signficances are computed by
+    permuting the residuals.
+
+    Parameters
+    ----------
+    y : DistanceMatrix, array_like, or filepath
+        Response distance matrix
+    x1, x2, ... : DistanceMatrix, array_like, or filepath
+        to distance matrices.  These distance matrices are predictors
+        for the regression.  They are also known as covariates
+    permutations : int, optional
+        Number of times to randomly permute `x` when assessing statistical
+        significance. Must be greater than or equal to zero. If zero,
+        statistical significance calculations will be skipped and the p-value
+        will be ``np.nan``.
+    random_state: int or np.RandomState, optional
+        Pseudo number generator state used for random sampling.
+    strict : bool, optional
+        If ``True``, raises a ``ValueError`` if IDs are found that do not exist
+        in all of the distance matrices. If ``False``, any nonmatching IDs are
+        discarded before running the test. See `n` (in Returns section below)
+        for the number of matching IDs that were used in the test. This
+        parameter is ignored if `x` and `y` are ``array_like``.
+    lookup : dict, optional
+        Maps each ID in the distance matrices to a new ID. Used to match up IDs
+        across distance matrices prior to running the Mantel test. If the IDs
+        already match between the distance matrices, this parameter is not
+        necessary. This parameter is disallowed if `x` and `y` are
+        ``array_like``.
+
+    Returns
+    -------
+    B : np.array
+        Array of coefficients
+    T : np.array
+        Array of pseudo t-statistics for each coefficient
+    pvals : np.array
+        Array of p-values for each coefficient calculated
+        using pseudo T-tests
+    F : float
+        pseudo F-statistic for lack of fit
+    model_pval : float
+        pvalue from pseudo F-test for lack of fit
+    R2: float
+        Coefficient of determination squared
+
+    See Also
+    --------
+    DistanceMatrix
+
+    References
+    ----------
+    .. [1] Legendre, P. and Legendre, L. (2012) Numerical Ecology. 3rd English
+       Edition. Elsevier.
+    .. [2] Legendre, P. Lapointe, F., Casgrain P. (1994) Modeling Brain
+       Evolution from Behavior: A Permutational Regression Approach
+    .. [3] https://cran.r-project.org/web/packages/ecodist/index.html
+    """
+
+    # Unpack kwargs
+    params = {'permutations': 10000,
```

Comment:
:+1: to 999 for consistency. Also, 999 makes sense as a default because you're computing the p-value as `(count_extreme + 1) / (permutations + 1)` (i.e., counting the *original* test statistic as one of the permutations).